src/createApp.ts

```diff
@@ -402,35 +453,51 @@ const createApp = compose({
 
 			return promise;
 		});
+		const instanceHandle = (<App> this)._instanceRegistry.addAction(action, id);
 
 		return {
 			destroy() {
 				this.destroy = noop;
+				instanceHandle.destroy();
 				registryHandle.destroy();
 			}
 		};
 	},
 
 	registerActionFactory(id: Identifier, factory: ActionFactory): Handle {
+		let destroyed = false;
+		let instanceHandle: Handle;
 		let registryHandle = actions.get(this).register(id, () => {
 			const promise = Promise.resolve().then(() => {
 				// Always call the factory in a future turn. This harmonizes behavior regardless of whether the
 				// factory is registered through this method or loaded from a definition.
 				return factory(this._registry);
+			}).then((action) => {
```

Comment:
I have seen this a few places, and I was ignoring it, but I figured we should at least mention it and discuss...  I don't think we have been explicit about it in the style guide, but the Dojo 2 convention had been with chained functions, that like `else` and other blocks, we would start it on a new line.  The convention made it easier to "copy pasta" chains, but also, I personally think it increases readability...  I guess we should really clarify it in the style guide, but we do have at least two different conventions in our code base now...  Example:

```typescript
const promise = Promise.resolve()
    .then(() => {
        /* code */
    })
    .then((action) => {
        /* code */
    });
```